lisa/sut_orchestrator/azure/features.py

```diff
@@ -563,6 +563,11 @@ def is_over_nd(self) -> bool:
         dmesg = self._node.tools[Dmesg]
         return "hvnd_try_bind_nic" in dmesg.get_output()
 
+    def _is_legacy_device(self) -> bool:
+        lspci = self._node.tools[Lspci]
```

Comment:
Why is this method implemented in Azure? Shouldn't be it independent with cloud?

Reply:
I wasn't sure whether this is cloud independent. Other clouds may use other legacy cards such as ConnectX-IB which I don't think is used in Azure. But I guess the same logic could still be used across all clouds. I can move this logic to infiniband.py.